enterprise/models.py

```diff
@@ -304,6 +305,54 @@ def __repr__(self):
         return self.__str__()
 
 
+@python_2_unicode_compatible
+class PendingEnrollment(TimeStampedModel):
+    """
+    Track future enrollments for PendingEnterpriseCustomerUser.
+
+    Store a course ID, an intended enrollment mode, and a link to a PendingEnterpriseCustomerUser;
+    when the PendingEnterpriseCustomerUser is converted to a full EnterpriseCustomerUser, API
+    calls will be made to enroll the newly-created user in whatever courses have been added.
+    """
+
+    user = models.ForeignKey(
+        PendingEnterpriseCustomerUser,
+        null=False,
+    )
+    course_id = models.CharField(
+        max_length=255,
+        blank=False,
+    )
+    course_mode = models.CharField(
+        max_length=25,
+        blank=False
+    )
+
+    class Meta(object):
+        unique_together = (("user", "course_id"),)
+
+    def complete_enrollment(self):
+        """
+        Enroll a newly-create user in the relevant course.
+        """
+        user = User.objects.get(email=self.user.user_email)
```

Comment:
No; I don't think there's anything we could do here that'd be more helpful, diagnostically-speaking, than the DoesNotExist that would happen if `complete_enrollment` was called on a PendingEnrollment that didn't have a real user yet.